src/node/plugin.ts

```diff
@@ -1,13 +1,40 @@
-import { Logger, field } from "@coder/logger"
+import { field, Level, Logger } from "@coder/logger"
 import * as express from "express"
 import * as fs from "fs"
 import * as path from "path"
 import * as semver from "semver"
 import * as pluginapi from "../../typings/pluginapi"
+import { HttpCode, HttpError } from "../common/http"
 import { version } from "./constants"
+import { replaceTemplates } from "./http"
+import { proxy } from "./proxy"
 import * as util from "./util"
+import { Router as WsRouter, WebsocketRouter, wss } from "./wsRouter"
 const fsp = fs.promises
 
+/**
+ * Inject code-server when `require`d. This is required because the API provides
+ * more than just types so these need to be provided at run-time.
+ */
+const originalLoad = require("module")._load
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+require("module")._load = function (request: string, parent: object, isMain: boolean): any {
```

Comment:
What does this actually return? Would it be safer to add that as the type instead of `any`? Or maybe we can create something like:
```typescript
type FixMeLater = any
```

Reply:
`_load` is used by `require` to load a module which can actually be anything so unfortunately we can't narrow down the type.

Reply:
But yeah definitely down to use a type alias for the `any`.